keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,687 @@
+admin namespace label.
+
+### Kube-controller-manager Changes
+
+// TODO: what part of claim.Status.Allocation should be updated? e.g.
+AdminAccess is part of `DeviceRequestAllocationResult` but need to set it for
+each device?
+
+In pkg/controller/resourceclaim/controller.go, process requests in `handleClaim`
+function to prevent creation of `ResourceClaim` when the `ResourceClaimTemplate`
+has the `adminAccess` field while the feature gate is turned off.
+
+### Kube-scheduler Changes
+
+In pkg/scheduler/framework/plugins/dynamicresources/allocateddevices.go, handle
+claims with `adminAccess` to ensure devices allocated with `adminAccess` are
+skipped without invoking the callback bypassing standard allocation checks.
+
+### ResourceQuota
+
+Requests asking for `adminAccess` contribute to the quota. We recommend that
+these administrator namespaces should not have quotas configured.
+
+### Test Plan
+
+<!--
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes
+necessary to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+- `<package>`: `<date>` - `<test coverage>`
+-->
+
+Start of v1.33 development cycle (v1.33.0-alpha.1-):
+
+- `k8s.io/kubernetes/pkg/registry/resource/resourceclaim`: %
+- `k8s.io/kubernetes/pkg/registry/resource/resourceclaimtemplate`: %
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+End-to-end testing depends on a working resource driver and a container runtime
+with CDI support. A
+[test driver](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/dra/test-driver)
```

Comment:
@pohly do we want to ask for implementations in the kubernetes-sigs/dra-example-driver? Or is that best effort in the context of the KEP process?

Reply:
That is a good criteria for beta graduation, but not for E2E testing.